### ISSUE TITLE
arrange method does not fetch parents

### ISSUE THREAD

**Issue by antonovga** [stefankroes/ancestry#164]
tag_root = Tag.create(name: "foo")
tag_child = t.children.create(name: "bar")

Tag.where(id: tag_child.id).arrange returns only tag_child - { tag_child => { } } 
expected to return: 
{ tag_root => { tag_child } }

Is it expected behaviour?

How can I fetch a hash with parents?

**Comment by caherrerapa** [comment]
Any ideas on this one? i'm trying to achieve the same but don't want to iterate to build the hash

**Comment by kbrock** [comment]
Will `Tag.where(id: tag_child.id).subtree.arrange` do what you need?

**Comment by caherrerapa** [comment]
@kbrock my case is the following
|--1
|--2
|--3
    |--4
        |---5

i have a query that returns 5 (leaf) and 2 (root) how can i have an arrange that has 2 and 3, 4, 5?

**Comment by kbrock** [comment]
I am guessing you mean:

```
|--1 **N**
|--2  **Y**
  |--3  **Y**
      |--4 **Y**
          |---5 **Y**
```

I expect the following to work:

```ruby
Tag.where(id: 2).subtree.arrange
```

Did I interpret this correctly?

**Comment by caherrerapa** [comment]
Not really, 2 is not the parent of 3. In my case 2 is a root node. I want to be able arrange getting 2, 3, 4, 5  but arrange only receive a single subtree.

I m using the following:

```
Category.where(id: Product.accessible_for(customer).pluck(:category_id))
                          .map(&:path_ids)
                          .flatten
```

**Comment by kbrock** [comment]
Pretty sure recent versions of arrange will meet your needs. (not just the current one)
We added a few pruning options
